core/modules/simpletest/backdrop_web_test_case.php

```diff
@@ -1217,34 +1225,44 @@ protected function backdropCreateContentType($settings = array()) {
    *   List of files that match filter.
    */
   protected function backdropGetTestFiles($type, $size = NULL) {
-    if (empty($this->generatedTestFiles)) {
-      // Generate binary test files.
-      $lines = array(64, 1024);
-      $count = 0;
-      foreach ($lines as $line) {
-        simpletest_generate_file('binary-' . $count++, 64, $line, 'binary');
-      }
+    $files = array();
+    // Make sure type is valid.
+    if (in_array($type, array('binary', 'html', 'image', 'javascript', 'php', 'sql', 'text'))) {
 
-      // Generate text test files.
-      $lines = array(16, 256, 1024, 2048, 20480);
-      $count = 0;
-      foreach ($lines as $line) {
-        simpletest_generate_file('text-' . $count++, 64, $line, 'text');
-      }
+      if (!in_array($type, $this->generatedTestFiles)) {
```

Comment:
Smart. I like how this only generates the individual test file requested instead of generating all of them.